From 7495a300915bda49261c5d8fb7c93ca563d5c424 Mon Sep 17 00:00:00 2001
From: Icenowy Zheng <zhengxingda@iscas.ac.cn>
Date: Tue, 2 Jun 2026 16:14:46 +0800
Subject: [PATCH 239/263] arm64: dts: qcom: sc8280xp-crd: set GPI DMA channels

It's found that some GPI interfaces shouldn't be touched by HLOS,
otherwise system hang / reset happens.

The GPI interface availability information is available via the ACPI
DSDT table, as interrupt resources for \_SB_.QGP* devices.

Add such information to the device tree file.

The DSDT table is acquired by extraction of qcfirmware8280_NVME.cab file
version 200.0.68.0 from Windows Update.

Signed-off-by: Icenowy Zheng <zhengxingda@iscas.ac.cn>
Link: https://lore.kernel.org/r/20260602081451.3808833-2-zhengxingda@iscas.ac.cn
---
 arch/arm64/boot/dts/qcom/sc8280xp-crd.dts | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/arch/arm64/boot/dts/qcom/sc8280xp-crd.dts b/arch/arm64/boot/dts/qcom/sc8280xp-crd.dts
index 6d81cae56..df4f01276 100644
--- a/arch/arm64/boot/dts/qcom/sc8280xp-crd.dts
+++ b/arch/arm64/boot/dts/qcom/sc8280xp-crd.dts
@@ -491,14 +491,17 @@ &dispcc0 {
 };
 
 &gpi_dma0 {
+	dma-channel-mask = <0x3>;
 	status = "okay";
 };
 
 &gpi_dma1 {
+	dma-channel-mask = <0x3>;
 	status = "okay";
 };
 
 &gpi_dma2 {
+	dma-channel-mask = <0x3>;
 	status = "okay";
 };
 
-- 
2.43.0

